map/src/components/map.tsx

```diff
@@ -165,6 +165,44 @@ function infoWindoContent(info: MarkerInfo): string {
   </div>`;
 }
 
+/**
+ * limits your function to be called at most every W milliseconds, where W is wait.
+ * Calls over W get dropped.
+ * Thanks to Pat Migliaccio.
+ * see https://medium.com/@pat_migliaccio/rate-limiting-throttling-consecutive-function-calls-with-queues-4c9de7106acc
+ * @param fn
+ * @param wait
+ * @example let throttledFunc = throttle(myFunc,500);
+ */
+function throttle(fn: Function, wait: number) {
+  let isCalled = false;
+
+  return (...args: any[]) => {
+    if (!isCalled) {
+      fn(...args);
+      isCalled = true;
+      setTimeout(() => {
+        isCalled = false;
+      }, wait);
+    }
+  };
+}
+
+const throttledReplaceHistory = throttle((map: google.maps.Map) => {
```

Comment:
It'd be better to use a debounce rather than a throttle here because we don't want to loose the last call to this function, as that is the one we care about, and may potentially be quite different to a previous call.

Additionally we have `lodash` as a dependency, so we can just use the function defined there:
https://lodash.com/docs/4.17.15#debounce